src/Engine/ProtoCore/Reflection/MirrorData.cs

```diff
@@ -230,10 +232,21 @@ public string StringData
                         // https://msdn.microsoft.com/en-us/library/3hfd35ad(v=vs.110).aspx
                         // We should always use invariant culture format for formattable 
                         // object.
-                        return (Data as IFormattable).ToString(null, CultureInfo.InvariantCulture);
+                        if (Data is double)
+                        {
+                            return (Data as IFormattable).ToString(PrecisionFormat, CultureInfo.InvariantCulture);
+                        }
+                        else
```

Comment:
I would comment this code for the time being and leave a TODO comment here saying "uncomment this once https://jira.autodesk.com/browse/DYN-5101 is complete".